src/metabase/settings/models/setting.clj

```diff
@@ -166,11 +166,18 @@
            (or (:setting/disabled-reasons (ex-data e))
                (throw e))))))
 
+(defn- warning?
```

Comment:
it's an interesting use of namespaced keyword. but wdyt about having a hierarchy here?